### ISSUE TITLE
Faulty interaction between recipes and scene dimension

### ISSUE THREAD

**Issue by termi-official** [MakieOrg/Makie.jl#3925]
- [x] are you running newest version (version from docs) ?
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [x] What platform + GPU are you on? Arch Linux + RTX 2080

When using recipes that could either yield a 2D or 3D plot (e.g. Linesegments) it now always renders the 2D version, no matter if I pass 2D or 3D data. MWE

```julia
using Makie, WGLMakie, Bonito

Page()

struct MakiePlotter{dim}
end

@recipe(MyLines) do scene
    Attributes(
        plotnodes=true,
    )
end


function Makie.plot!(WF::MyLines{<:Tuple{<:MakiePlotter{dim}}}) where dim
    plotter = WF[1][]
    if dim == 2
        lines = [Point2f(0.0,0.0)]
    else
        lines = [Point3f(1.0,1.0,0.0), Point3f(0.0,0.0,1.0)]
        @info "Peekaboo"
    end
    Makie.scatter!(WF, lines)
end

mylines(MakiePlotter{2}())  # 2D
mylines(MakiePlotter{3}())  # 2D
scatter([Point2f(0.0,0.0)]) # 2D
scatter([Point3f(1.0,1.0,0.0), Point3f(0.0,0.0,1.0)]) # 3D
```

I added the logging statement to make sure that the code is really executed.

Versions info:
```
  [824d6782] Bonito v3.1.1
  [ee78f7c6] Makie v0.21.2
  [276b4fcb] WGLMakie v0.10.2
```

**Comment by asinghvi17** [comment]
Yes, I think this system was refactored in v0.21.  In v0.20 and before, a plot would be executed once in an empty Scene to determine its dimension (2d or 3d), and then plotted into an `Axis` if 2D or a `LScene` if 3D.

In 0.21, this behaviour was changed to use the `preferred_axis_dimension` and `args_preferred_axis` functions, see here:
https://github.com/MakieOrg/Makie.jl/blob/37f58686e1cae4bafeda30744b53078fd0723c87/src/figureplotting.jl#L95-L108
(`preferred_axis_dimension` calls `args_preferred_axis` in its generic implementation).

The solution to this MWE would be to define:
```julia
Makie.args_preferred_axis(::Type{<: MyLines}, ::MakiePlotter{Dim}) = Dim == 2 ? Makie.Axis : Makie.LScene
```

This needs documentation - and we should probably overhaul the recipe tutorial again.

**Comment by termi-official** [comment]
Let me fix it then.
